flow/layers/image_filter_layer.cc

```diff
@@ -11,6 +11,43 @@ ImageFilterLayer::ImageFilterLayer(sk_sp<SkImageFilter> filter)
       transformed_filter_(nullptr),
       render_count_(1) {}
 
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+void ImageFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  DiffContext::AutoSubtreeRestore subtree(context);
+  auto* prev = static_cast<const ImageFilterLayer*>(old_layer);
+  if (!context->IsSubtreeDirty()) {
+    FML_DCHECK(prev);
+    if (filter_ != prev->filter_) {
+      context->MarkSubtreeDirty(context->GetOldLayerPaintRegion(old_layer));
+    }
+  }
+
+  DiffChildren(context, prev);
+
+  SkMatrix inverse;
+  if (context->GetTransform().invert(&inverse)) {
+    auto paint_bounds =
+        inverse.mapRect(context->CurrentSubtreeRegion().GetBounds());
+
+    auto filter_bounds =
+        filter_->filterBounds(paint_bounds.roundOut(), SkMatrix::I(),
+                              SkImageFilter::kForward_MapDirection);
+    context->AddPaintRegion(SkRect::Make(filter_bounds));
+
+    // Technically, there is no readback with ImageFilterLayer, but we can't
+    // clip the filter (because it may sample out of clip rect) so if any part
+    // of layer is repainted the whole layer needs to be.
+    // TODO(knopp) There is a room for optimization here - this doesn't need to
+    // be done if we know for sure that we're using raster cache
+    context->AddReadbackRegion(SkRect::Make(filter_bounds));
```

Comment:
> which should be appropriately expanded ignoring the clip

If this is indeed the case then this is unnecessary. Once I get through this pass I plan to rebase the rest of partial update code so this can be tested.

Reply:
The question here is, whether `SkCanvas::quickReject` takes into account expanded clip for saved layer. If it does, I think this code can be removed. If it doesn't, it needs to be there because layers that are out of clip but affect the blurred content will be rejected in `Layer::needs_painting()`.